tests/base/utilities_10.cc

```diff
@@ -0,0 +1,76 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2015 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+// test Utilities::split_string_list with a string that only contains
+// the delimiter and, possibly, spaces
+
+#include "../tests.h"
+#include <iomanip>
+#include <iomanip>
+#include <fstream>
+#include <cmath>
+
+#include <deal.II/base/utilities.h>
+
+
+void test ()
+{
+  // verify the documented behavior of eating trailing delimiters
+  {
+    deallog << Utilities::split_string_list (",").size()
+	    << std::endl;
+    deallog << Utilities::split_string_list (" , ").size()
+	    << std::endl;
+  }
+
+  {
+    deallog << Utilities::split_string_list (",,").size()
+	    << std::endl;
+    deallog << Utilities::split_string_list (" , , ").size()
+	    << std::endl;
+  }
+
+  // try some more esoteric cases:
+  {
+    deallog << Utilities::split_string_list (" , , ", ' ').size()
+	    << std::endl;
+  }
+
+  {
+    deallog << Utilities::split_string_list (" ", ' ').size()
+	    << std::endl;
+    deallog << Utilities::split_string_list ("   ", ' ').size()
+	    << std::endl;
+  }
+
+  Assert (Utilities::split_string_list(" ; ", ';').size() == 1,
+	  ExcInternalError());
+  Assert (Utilities::split_string_list(" ; ", ';')[0] == "",
+	  ExcInternalError());
```

Comment:
Do you want to use Assert or AssertThrow ?  
I checked in the testsuite and we use Assert 2644 times and AssertThrow 63 times. I found several tests that do nothing in release in mode :( Should we just change all of the Assert to AssertThrow or go through all the tests and see what's appropriate ?

Otherwise it looks good, you can merge it.